src/MCPServer/lib/server/rpc_server.py

```diff
@@ -460,23 +475,26 @@ def _pull_choices(job_id, lang, jobs_awaiting_for_approval):
     """
     ret = {}
     try:
-        choices = jobs_awaiting_for_approval[job_id].choices
+        choices = jobs_awaiting_for_approval[job_id].get_choices()
     except (KeyError, AttributeError):
         raise JobNotWaitingForApprovalError
-    for item in choices:
-        id_, label, rd = item
+    for item in choices.items():
+        id_, label = item
         ret[id_] = label[lang]
     return ret
 
 
-def start(workflow):
-    worker = RPCServer(workflow)
+def start(workflow, shutdown_event, package_queue):
+    worker = RPCServer(workflow, shutdown_event, package_queue)
```

Comment:
Hey @cole, I've seen a `MySQL server has gone away` today after 8+ hours running your branch idle (see https://gist.github.com/sevein/266f8a2e71af2a1f8a527c8900832f52). Using the application defaults. MySQL's wait_timeout equals 28800. I can count up to two places in the same thread where we close old connections in that code path: the RPC handler wrapper plus create_package - not sure what's going on!

Reply:
Looking at the traceback I think I know what is happening here — the key is that db connections in the Django ORM are per thread, and the RPC handler wrapper and create pacakge will be executed in one of the RPCServer threads, but then create package runs `_start_package_transfer_with_auto_approval` using the executor, meaning it runs on a different thread. It looks like in that chain, there is no close old connections. I'll add some more.

Reply:
OK, hopefully fixed now!